App/ViewModels/TrayWindowViewModel.cs

```diff
@@ -23,22 +25,46 @@ public partial class TrayWindowViewModel : ObservableObject
 
     private DispatcherQueue? _dispatcherQueue;
 
-    [ObservableProperty] public partial VpnLifecycle VpnLifecycle { get; set; } = VpnLifecycle.Unknown;
+    [ObservableProperty]
+    [NotifyPropertyChangedFor(nameof(ShowEnableSection))]
+    [NotifyPropertyChangedFor(nameof(ShowWorkspacesHeader))]
+    [NotifyPropertyChangedFor(nameof(ShowNoAgentsSection))]
+    [NotifyPropertyChangedFor(nameof(ShowAgentsSection))]
+    [NotifyPropertyChangedFor(nameof(ShowFailedSection))]
```

Comment:
ShowFailedSection doesn't depend on the lifecycle.

It really is annoying that we have to maintain these deps this way, where you have to mark the dependency, not the dependent. Harder to verify it's correct.

Reply:
Yeah I definitely don't like it either. Since it's a decorator attribute I wonder if someone could write an inverse version of this using codegen or something. Maybe one already exists and I've just been doing it the dumb way

Reply:
Thanks for spotting this though!